modules/monitor/src/components/Monitor.vue

```diff
@@ -245,160 +265,242 @@ export default {
   },
   data() {
     return {
-      op: this.$api.qsub("query network:tree/").prepFn("", function(key, obj) {
+      op: this.$api.qsub("query network:tree/ where Hidden is false").prepFn("", function(key, obj) {
         if (!obj._key) {
           obj._key = key;
 
-          if (obj.VirtualProcess) {
+          if (obj.Virtual) {
             obj._treeLayer = -1; // -1, 1, 2, 3
-            obj._parentKey = `network:tree/${obj.ParentPid}`;
+            obj._processKey = `network:tree/${obj.ParentPid}`;
           } else {
             switch (countChar(key, "/")) {
               case 1:
                 // process
                 obj._treeLayer = 1;
+                obj.lastActivity = 0;
                 break;
               case 2:
-                // communication
+                // dns connection
                 obj._treeLayer = 2;
-                obj._parentKey = key.split("/").slice(0, 2).join("/");
+                obj._processKey = key
+                  .split("/")
+                  .slice(0, 2)
+                  .join("/");
+                obj._scopeKey = obj._key;
+                obj.lastActivity = 0;
                 break;
               case 3:
-                // link
+                // connection
                 obj._treeLayer = 3;
-                obj._parentKey = key.split("/").slice(0, 3).join("/");
+                obj._processKey = key
+                  .split("/")
+                  .slice(0, 2)
+                  .join("/");
+                obj._scopeKey = key
+                  .split("/")
+                  .slice(0, 3)
+                  .join("/");
                 break;
               default:
                 console.log(`WARNING: unexpected count of / in key ${key}`);
             }
           }
         }
       }),
+      opChanged: this.$api.info().changeCnt,
+      treeCache: null,
       selected: 0,
       selectedProcess: null,
-      selectedCommunication: null,
-      treeCache: [],
-      unconnectedRecords: 0,
-      firstRun: true
+      selectedScope: null
     };
   },
   computed: {
-    anyActivity() {
-      for (var i = 0; i < this.tree.length; i++) {
-        if (this.tree[i]._children && this.tree[i]._children.length > 0) {
-          return true;
-        }
-      }
-      return false;
-    },
     tree() {
```

Comment:
This function is very very long, can we split it up?

Reply:
Will have to look into this later, fast fixes would only make it minimally better.